docs/_ext/codediff.py

```diff
@@ -95,12 +96,24 @@ class CodeDiffDirective(SphinxDirective):
     }
 
   def run(self):
-    new_content = CodeDiffParser().parse(list(self.content), **self.options)
-
-    node = nodes.paragraph()
-    self.content = ViewList(new_content, self.content.parent)
-    self.state.nested_parse(self.content, self.content_offset, node)
-    return [node]
+    table_code, test_code = CodeDiffParser().parse(list(self.content), **self.options)
+
+    # Create a test node as a comment node so it won't show up in the docs.
+    # We add attribute "testnodetype" so it is be picked up by the doctest
```

Comment:
Is there a link to Sphinx documentation that describes this `testnodetype='testcode'` trick?

Reply:
It is not part of the documentation, but I found it out by going through the source code. I've added some explanation in this comment to clarify that.